Quaver.Shared/Online/MultiplayerGameExtensions.cs

```diff
@@ -21,19 +22,35 @@ public static class MultiplayerGameExtensions
         /// <returns></returns>
         public static string GetDifficultyName(this MultiplayerGame game)
         {
-            var diffName = "";
-            var pattern = @"\[(.*?)\]";
-            var matches = Regex.Matches(game.Map, pattern);
+            if (game.Map.LastOrDefault() != ']')
+            {
+                // Huh?
+                Logger.Warning($"Multiplayer map name not ending in ']': {game.Map}", LogType.Runtime);
+                return "";
+            }
 
-            foreach (Match match in matches)
+            // The map string has a form of "artist - title [difficulty]". Look for the first " [" after the first " - ".
+            //
+            // This isn't quite perfect (it's possible trick it by having [, ] or - in a particular order in
+            // artist, name or difficulty name), but it works decently well and handles [something] inside
+            // a difficulty name.
+            var dash = game.Map.IndexOf(" - ");
+            if (dash == -1)
             {
-                if (match != matches.Last())
-                    continue;
+                // Shouldn't happen.
+                Logger.Warning($"Multiplayer map name doesn't contain \" - \": {game.Map}", LogType.Runtime);
+                return "";
+            }
 
-                diffName = match.Groups[1].ToString();
+            var spaceBracket = game.Map.IndexOf(" [", dash + 3);
+            if (spaceBracket == -1)
+            {
+                // Also shouldn't happen.
+                Logger.Warning($"Multiplayer map name doesn't contain \" [\" after \" - \": {game.Map}", LogType.Runtime);
+                return "";
             }
 
-            return diffName;
+            return game.Map.Substring(spaceBracket + 2, game.Map.Length - spaceBracket - 3);
```

Comment:
This is something that could be solved by making the regex more robust.

Try: `/.* - .* \[(.*)\]$/`
https://regex101.com/r/Mntx07/1

Does this work?

Reply:
So the original diffs I noticed the issue on look like this: `artist - title [difficulty [yes]]` and both this regex and some variations I'm trying (`^.+ - .+ \[(.+)\]$`) fail the same way on it. `title [difficulty` becomes the title and `yes]` the difficulty.

That isn't to say the code here doesn't have failure modes (`artist - title [title] [difficulty]` is one of them); really the best solution would be to send the strings separately.

Reply:
True. Gonna close this one then & fix server side